django/db/models/options.py

```diff
@@ -230,6 +231,9 @@ def contribute_to_class(self, cls, name):
                 self.db_table, connection.ops.max_name_length()
             )
 
+        if self.swappable:
+            setting_changed.connect(self.setting_changed)
```

Comment:
Cool that was my main concern when I saw the ticket.

Since `weak` defaults to `True` we'll make sure not to keep a strong reference to all the model classes created during migrations (hundreds of them in tests).

Reply:
Yup. I considered being a little more efficient by having a single, batch signal receiver that maps settings to “live” swappable models. But I don’t think that’s worth it as most projects use, at most, one swappable model (`auth.User`).